elixir/markdown/lib/markdown.ex

```diff
@@ -11,37 +11,53 @@ defmodule Markdown do
     "<h1>Header!</h1><ul><li><em>Bold Item</em></li><li><i>Italic Item</i></li></ul>"
   """
 
-  #This should probably be a pipeline instead of all those nested functions
+  #Refactored into a pipeline to get rid of the nested functions
   @spec parse(String.t()) :: String.t()
   def parse(m) do
-    patch(Enum.join(Enum.map(String.split(m, "\n"), fn t -> process(t) end)))
-  end
+    m
+    |> String.split("\n")
+    |> IO.inspect(label: "output of String.split")
+    |> Enum.map(& process(&1))
+    |> IO.inspect(label: "Enum.map process output")
+    |> Enum.join()
+    |> IO.inspect(label: "output of Enum.join")
```

Comment:
Do you want to remove the `IO.inspect`s? Using a tool like [Credo](https://github.com/rrrene/credo) can automate these checks for you.

Reply:
You can respond to the above comment or click "resolve conversation" to collapse this thread!

Reply:
There should be no more IO.inspects now

Reply:
:+1: